### ISSUE TITLE
Getting timeout errors in cy.writeFile Randomly during automation runs - cy.writeFile() timed out after waiting `4000ms`.

### ISSUE THREAD

**Issue by laksman2003** [cypress-io/cypress#20420]
### Current behavior

 cy.writeFile() randomly throws timeout error.

### Desired behavior

 cy.writeFile() should write the file if called multiple times without any flakiness.

### Test code to reproduce

```
cy.writeFile(fileName, 'Tests Started Running\n');

cy.wrap(null).then(() => {cy.writeFile(fileName, "Test Step ..."+ ".Passed. \n", {
flag: 'a+'
})
})
```

### Cypress Version

9.4.1

### Other

Docker Image - FROM cypress/included:9.4.1

Below is the code snippet from it block. Need to understand the root cause of this random failure, I also tried increasing the timeout and added the encoding as utf-8 but still no luck.

```
cy.writeFile(fileName, 'Tests Started Running\n');

cy.wrap(null).then(() => {cy.writeFile(fileName, "Test Step ..."+ ".Passed. \n", {
flag: 'a+'
})
})
```

**Comment by laksman2003** [comment]
@jennifer-shehane Is there any workaround or update for this issue?

**Comment by FilipKelemen** [comment]
I have the same issue. Happens usually in bitbucket pipelines and browserstack runs, never encountered using "cypress open", basically only cypress run. Also, 7.7.0 version of cypress makes the timeout go away, 9.4.0 is the version where we cy.writeFile times out.

**Comment by FilipKelemen** [comment]
> I have the same issue. Happens usually in bitbucket pipelines and browserstack runs, never encountered using "cypress open", basically only cypress run. Also, 7.7.0 version of cypress makes the timeout go away, 9.4.0 is the version where we cy.writeFile times out.

I also noticed this started happening after using fs-extra in cypress/plugins/index.ts file

**Comment by FilipKelemen** [comment]
I think the issue is actually that cy.writeFile looks like it's timing out if the file does not exist, this did not happen before

**Comment by FilipKelemen** [comment]
> I have the same issue. Happens usually in bitbucket pipelines and browserstack runs, never encountered using "cypress open", basically only cypress run. Also, 7.7.0 version of cypress makes the timeout go away, 9.4.0 is the version where we cy.writeFile times out.

Solved it downgrading to cypress 7.7.0 for tests using cy.writeFile
